lib/config/fileconf.go

```diff
@@ -605,6 +606,48 @@ func (u *UniversalSecondFactor) Parse() (types.U2F, error) {
 	return res, nil
 }
 
+// AllowTCPForwarding describes the possible values of the
+// `allow_tcp_forwarding` SSH settings key. Essentially a wrapper around
+// regular.SSHPortForwardingMode to plug it into the YAML parser. This
+// allows us to define the parsing code _here_, rather than cluttering
+// up the SSH server code with irrelevant YAML handling.
+type AllowTCPForwarding regular.SSHPortForwardingMode
+
+// UnmarshalYAML parses a YAML representation of a SSHPortForwardingMode,
+// failing if the value cannot be converted.
+func (mode *AllowTCPForwarding) UnmarshalYAML(unmarshal func(interface{}) error) error {
```

Comment:
If we only support our boolean for this field would make the implementation much simpler. For example, you can drop this function.